x-pack/plugins/apm/server/plugin.ts

```diff
@@ -132,8 +134,18 @@ export class APMPlugin
               [PROCESSOR_EVENT]: {
                 type: 'keyword',
               },
+              [AGENT_NAME]: {
+                type: 'keyword',
+              },
+              [SERVICE_LANGUAGE_NAME]: {
+                type: 'keyword',
+              },
+              labels: {
+                type: 'object',
+                dynamic: true,
```

Comment:
dynamic defaults to `true` for the object field type so I don't think this is strictly necessary?

Reply:
When i take away `dynamic: true`, my aggregation fails:
```
GET .alerts-observability.apm.alerts-*/_search
{
  "size": 10,
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "kibana.alert.rule.producer": "apm"
          }
        },
        {
          "term": {
            "kibana.alert.status": "active"
          }
        }
      ]
    }
  },
  "aggs": {
    "service_groups": {
      "filters": {
        "filters": {
          "2641b250-5aba-11ed-a895-07dc9e2f77a8": {
            "bool": {
              "should": [
                {
                  "match": {
                    "labels.country_code": "US"
                  }
                }
              ],
              "minimum_should_match": 1
            }
          }
        }
      },
      "aggs": {
        "alerts_count": {
          "cardinality": {
            "field": "kibana.alert.uuid"
          }
        }
      }
    }
  }
}
```
Returns:
```
{
  "took": 1,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 6,
      "relation": "eq"
    },
    "max_score": 0,
    "hits": [
      {
        "_index": ".internal.alerts-observability.apm.alerts-default-000001",
        "_id": "ff1ef1f2-8ad7-4949-b610-c11bd5dfe2c7",
        "_score": 0,
        "_source": {
          "kibana.alert.rule.category": "Error count threshold",
          "kibana.alert.rule.consumer": "alerts",
          "kibana.alert.rule.execution.uuid": "46ecf66e-6c55-4806-89db-ca67ae659242",
          "kibana.alert.rule.name": "> 25 errors last 1 min",
          "kibana.alert.rule.producer": "apm",
          "kibana.alert.rule.rule_type_id": "apm.error_rate",
          "kibana.alert.rule.uuid": "090e77d0-50b1-11ed-8299-8da1b02a8d3b",
          "kibana.space_ids": ["default"],
          "kibana.alert.rule.tags": [],
          "@timestamp": "2022-11-06T16:07:54.794Z",
          "service.name": "elastic-co-frontend",
          "service.environment": "https://www.elastic.co",
          "processor.event": "error",
          "kibana.alert.evaluation.value": 29,
          "kibana.alert.evaluation.threshold": 25,
          "kibana.alert.reason": "Error count is 29 in the last 1 min for elastic-co-frontend. Alert when > 25.",
          "agent.name": "rum-js",
          "service.language.name": "javascript",
          "labels.in_eu": "false",
          "labels.country_code": "US",
          "labels.city": "council bluffs",
          "labels.ip": "35.224.28.11",
          "labels.lang": "en-US",
          "kibana.alert.duration.us": 312414000,
          "kibana.alert.time_range": {
            "gte": "2022-11-06T16:02:42.380Z"
          },
          "kibana.alert.instance.id": "apm.error_rate_elastic-co-frontend_https://www.elastic.co",
          "kibana.alert.start": "2022-11-06T16:02:42.380Z",
          "kibana.alert.uuid": "ff1ef1f2-8ad7-4949-b610-c11bd5dfe2c7",
          "kibana.alert.status": "active",
          "kibana.alert.workflow_status": "open",
          "event.kind": "signal",
          "event.action": "active",
          "tags": [],
          "kibana.version": "8.6.0"
        }
      },
      ...
    ]
  },
  "aggregations": {
    "service_groups": {
      "buckets": {
        "2641b250-5aba-11ed-a895-07dc9e2f77a8": {
          "doc_count": 0,
          "alerts_count": {
            "value": 0
          }
        }
      }
    }
  }
}
```
Notice how the document has field `"labels.country_code": "US"`, but the filters agg has doc_count: 0.

The mapping shows ` "labels": { "type": "object" },`:
```
{
  ".internal.alerts-observability.apm.alerts-default-000001": {
    "mappings": {
      "dynamic": "false",
      "_meta": {
        "namespace": "default",
        "kibana": { "version": "8.6.0" }
      },
      "properties": {
        "@timestamp": { "type": "date" },
        "agent": { "properties": {...} },
        "ecs": { "properties": {...} },
        "event": { "properties": {...} },
        "kibana": { "properties": {...} },
        "labels": { "type": "object" },
        "processor": { "properties": {...} },
        "service": {...},
        "tags": { "type": "keyword" },
        "transaction": { "properties": {...} }
      }
    }
  }
}
```
But keeping the `dynamic: true` here, allows the labels.country_code field to be accessible to the filters agg as expected.

Reply:
@ymao1 `dynamic` is inherited from its parents, and only if it is entirely omitted from the mapping, including at the top level, the value defaults to `true`.

Reply:
Gotcha, thanks! I'm fine with setting this to true if the top level setting is changed back to `strict`